### ISSUE TITLE
NewVIdeoPostのview側の管理について。

### ISSUE THREAD

**Issue by MOTOMUR** [MOTOMUR/plz_donation#40]
YouTubeの動画投稿をこのサイトでも投稿するという連携機能について相談です。

現在の予定では、postsにYoutubeの時だけ、bodyに特殊なIDを書き込み、描画時に、bodyがIDと一致したら、video postとして、iframeとかを用いて、描画しようと思ってます。

なぜこのようにしたいかというと、postsに時系列順に、youtubeも普通のポストも分け隔てなく管理したいからという点と、

youtubeから送られてくる内容にchannelidはあるが、動画の説明文は持ってこれないため、余ってしまうbodyを使ってしまおう。という理由です。

現在のポストのルールに乗っかりつつ、動画投稿とただの投稿の描画の区別はつけたいので、これが一番ベストだと思うのですが、どうでしょうか？

以下が今までpostsに送っていたものです。

```
var postData = {
              author: result[1],
              uid: result[0],
              body: ,
              title: VideoTitle,
              post_image: null,
              postId: newPostKey,
              postStar: {
                starCount:0,
                stars:{}
              },
              postComment:{
                commentCount : 0,
                comments:{}
              },
              timestamp:admin.database.ServerValue.TIMESTAMP
            };
```

ポストイメージは今回はいらないので、nullで他はそのままだと思います。
bodyの部分に数行のIDを書いて、if文で一致をとって区別しようと思っていたのですが、他にもっとスマートな方法がありますでしょうか？

**Comment by karamari** [comment]
別に悪くはないと思いますが、個人的には`type`みたいなフィールドを用意して、普通の投稿は`type: "text"`、youtubeは`type: "youtube_vid"`とするのがいいと思います。

**Comment by MOTOMUR** [comment]
そうですね。typeというフィールドを作り、テキストかYoutubeかといったデータを入れて、Youtubeに限りbodyはVideoIDとして機能作成していきます。

**Comment by karamari** [comment]
> Youtubeに限りbodyはVideoIDとして機能作成していきます。

できれば使い回さずに別のプロパティを使ったほうがいいと思います。
